### PR TITLE
S2U-31 5.1.4.2 Assignments: Grades in the main view of assignments 

### PR DESCRIPTION
**Pull request by JuanDavid102**
that have already been evaluated

### CODE REVIEW

File: assignment/api/src/resources/assignment.properties [hunk 1]
```diff
@@ -360,6 +360,9 @@ opedat = Open date for assignment ''{0}'' is {1}.
 
 passfail = Pass/Fail
 pass = Pass
+unchecked=Unchecked
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment.properties:363]: Sort by alphabetical order, please, or put the strings at the end of the file.

File: assignment/api/src/resources/assignment_es.properties [hunk 2]
```diff
@@ -362,6 +362,9 @@ opedat=Fecha de apertura de la tarea ''{0}'' en {1}.
 
 passfail=Aprobado / Suspendido.
 pass=Aprobada
+unchecked=Desmarcado
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_es.properties:365]: Sort by alphabetical order, please, or put the strings at the end of the file.

File: assignment/api/src/resources/assignment_eu.properties [hunk 3]
```diff
@@ -349,6 +349,9 @@ opedat=Irekitze-data ''{0}'' zereginerako {1} da.
 
 passfail=Gainditu/Ez gainditu
 pass=Gainditua
+unchecked=Markatu gabe
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_eu.properties:352]: Sort by alphabetical order, please, or put the strings at the end of the file.

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 4]
```diff
@@ -2983,6 +2983,14 @@ public String getMaxPointGradeDisplay(int factor, int maxGradePoint) {
         return getGradeDisplay(Integer.toString(maxGradePoint), Assignment.GradeType.SCORE_GRADE_TYPE, factor);
     }
 
+    public String getFactoredGrade(int factor, String grade) {
+        String factoredGrade = "0,00";
+        if (grade != null){
```
Comment (mpellicer) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:2988]: spacing

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 5]
```diff
@@ -2983,6 +2983,14 @@ public String getMaxPointGradeDisplay(int factor, int maxGradePoint) {
...
+            factoredGrade = (grade.length() == 1? "0,0" + grade: (grade.length() == 2? "0," + grade : grade.substring(0, grade.length()-2) + "," + grade.substring(grade.length()-2, grade.length())));
```
Comment (mpellicer) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:2989]: This looks a bit odd to me, could you explain it to me? AFAIK I see "0,00" as a potential grade but this will only work with some languages like ES.

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 6]
```diff
@@ -1321,6 +1321,8 @@ public String buildMainPanelContext(VelocityPortlet portlet, Context context, Ru
         } catch (IdUnusedException iue) {
             log.warn(this + ":buildMainPanelContext: Site not found!" + iue.getMessage());
         }
+        User user = (User) state.getAttribute(STATE_USER);
+        context.put("isStudent", s.getUserRole(user.getId()).getId().equals("Student"));
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1325]: The role could be different than "Student" in project sites or other custom site types, we need a different approach

Comment (JuanDavid102) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1325]: Ok. I will see about something that could work

Comment (jesusmmp) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1325]: I think the correct permission should be checked instead of role.

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm [hunk 7]
```diff
@@ -491,6 +496,28 @@
...
+											<span>$tlang.getString("gen.checked")</span>
+										#elseif ($submission.grade.equals("Unchecked"))
+											<span>$tlang.getString("unchecked")</span>
+										#elseif ($assignment.TypeOfGrade.ordinal() == 3)
+											<span>$service.getFactoredGrade($assignment.ScaleFactor,$submission.grade)/$service.getMaxPointGradeDisplay($assignment.ScaleFactor, $assignment.MaxGradePoint)</span>
+										#else 
+											<span>$submission.grade</span>
+										#end
+									#else
+										<span>-#if ($assignment.TypeOfGrade.ordinal() == 3)/$service.getMaxPointGradeDisplay($assignment.ScaleFactor, $assignment.MaxGradePoint)#end</span>
```
Comment (mpellicer) [inline assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: What is this '-' before the if?

**Review by mpellicer — CHANGES_REQUESTED** [review]
I feel this works fine in Spanish but could fail in other languages like English, can you confirm @JuanDavid102 ?

Good job, just some minor comments and concerns, keep rocking!

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm [hunk 8]
```diff
@@ -491,6 +496,28 @@
 									#end
 								#end
 							</td>
+							#if ($isStudent)
+								<td headers="grade">
+									#if ($service.getSubmissionStatus($submission.Id, false) == $tlang.getString("gen.returned"))
+										#if ($submission.grade.equals("pass"))
```
[the remark this answers is not in the document]
Comment (JuanDavid102) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:502]: Ok. I will Start putting the constant in the right side of the comparison. And after that i will try to create a new functions to this

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 6 above)
Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1325]: Exactly!

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm (hunk 7 above)
Comment (JuanDavid102) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: It's to get the -/100. When the teacher/instructor has not graded the assignments

**Comment by JuanDavid102** [comment]
Yes, don't work in English. I will modify it

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm (hunk 7 above)
Comment (mpellicer) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: From the user perspective, '-' can mean other things, better using a text like 'Not graded' 'Sin calificar' 'Sense qualificar' ?

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 6 above)
Comment (JuanDavid102) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1325]: Ok. I'll keep it in mind

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm (hunk 7 above)
Comment (jesusmmp) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: Maybe string 'Not graded' could be confused... student doesn't know if the grade is really not graded or unpublished...
Lo que para el estudiante es "Sin calificar", para el docente puede que no lo sea ya que puede estar "Calificada" pero sin publicar ¿me explico?

Comment (jesusmmp) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: Cuando la nota de la tarea está sin publicar, si el estudiante pulsa en la tarea ve

image

Yo personalmente prefiero que aparezca algo que se vea que dicha tarea esta sobre 10. De ahí el -/10

Comment (mpellicer) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: ¿Aceptamos guión entonces? So do we accept the '-' symbol?

Comment (jesusmmp) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:516]: Yes :)

**Comment by JuanDavid102** [comment]
Now works fine, and I clean the code

File: assignment/api/src/resources/assignment.properties [hunk 9]
```diff
@@ -275,7 +275,8 @@ gen.hideduedate = Hide due date from students
 grad2 = Grade entered ''{0}'' is larger than the maximum grade allowed ''{1}''. Please make a correction or click the original button again to save the entered value ''{0}''.
 
 grad3 = Graded
-
+grade.title=Grades
+grade.delimiter=.
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment.properties:279]: Never put delimiters into translation files.

Check the formattedText.getDecimalSeparator() method.

File: assignment/api/src/resources/assignment_ca.properties [hunk 10]
```diff
@@ -269,7 +269,8 @@ gen.hideduedate=Amaga als estudiants la data de lliurament
 grad2=La qualificaci\u00f3 introdu\u00efda \u2019{0}\u2019 \u00e9s m\u00e9s gran que la m\u00e0xima permesa \u2019{1}\u2019. Corregiu-ho o  feu clic altre cop sobre el bot\u00f3 per desar la qualificaci\u00f3 introdu\u00efda \u2019{0}\u2019.
 
 grad3=Qualificada
-
+grade.title=Qualificacions
+grade.delimiter=,
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_ca.properties:273]: Never put delimiters into translation files.

Check the formattedText.getDecimalSeparator() method.

File: assignment/api/src/resources/assignment_es.properties [hunk 11]
```diff
@@ -277,7 +277,8 @@ gen.hideduedate=Ocultar fecha de entrega a estudiantes
 grad2=La calificaci\u00f3n introducida, ''{0}'', es mayor que la puntuaci\u00f3n m\u00e1xima permitida, ''{1}''. Realice la correcci\u00f3n o, de lo contrario, vuelva a pulsar el bot\u00f3n para confirmar el valor introducido, ''{0}''.
 
 grad3=Corregida
-
+grade.title=Calificaciones
+grade.delimiter=,
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_es.properties:281]: Never put delimiters into translation files.

Check the formattedText.getDecimalSeparator() method.

File: assignment/api/src/resources/assignment_eu.properties [hunk 12]
```diff
@@ -266,7 +266,8 @@ gen.hideduedate=Ezkutatu emate-data ikasleei
 grad2=Sartutako nota, "{0}", gehienezko puntuazioa baino handiagoa da, hau da, "{1}". Zuzendu gaizki badago, bestela, gordetzeko botoia sakatu emandako nota berresteko, "{0}".
 
 grad3=Kalifikatua
-
+grade.title=Balorazioak
+grade.delimiter=,
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_eu.properties:270]: Never put delimiters into translation files.

Check the formattedText.getDecimalSeparator() method.

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 13]
```diff
@@ -2983,6 +2983,12 @@ public String getMaxPointGradeDisplay(int factor, int maxGradePoint) {
         return getGradeDisplay(Integer.toString(maxGradePoint), Assignment.GradeType.SCORE_GRADE_TYPE, factor);
     }
 
+    public String getFactoredGrade(int factor, String grade, String delimiter) {
```
Comment (mpellicer) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:2986]: What's the purpose of this method? What's the difference with 'getGradeDisplay'?

getGradeDisplay (grade, Assignment.GradeType.SCORE_GRADE_TYPE, factor)

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 14]
```diff
@@ -1321,7 +1321,7 @@ public String buildMainPanelContext(VelocityPortlet portlet, Context context, Ru
         } catch (IdUnusedException iue) {
             log.warn(this + ":buildMainPanelContext: Site not found!" + iue.getMessage());
         }
-
+
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:1324]: Revert this change please, it's empty

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm [hunk 15]
```diff
@@ -491,6 +496,24 @@
 									#end
 								#end
 							</td>
+							#if ($allowAddAssignment == false)
+								<td headers="grade">
+									#if ($service.getSubmissionStatus($submission.Id, false) == $tlang.getString("gen.returned") && $submission.grade)
+										#if ($assignment.TypeOfGrade.ordinal() == 2)
```
Comment (mpellicer) [inline assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:502]: I feel you're doing a similar operation than 'getGradeDisplay', can you confirm?

Comment (JuanDavid102) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_list_assignments.vm:502]: Yes, i could getting the grade with the "getGradeDisplay" function. Now, i change all

File: assignment/api/src/resources/assignment_eu.properties [hunk 16]
```diff
@@ -266,7 +266,7 @@ gen.hideduedate=Ezkutatu emate-data ikasleei
 grad2=Sartutako nota, "{0}", gehienezko puntuazioa baino handiagoa da, hau da, "{1}". Zuzendu gaizki badago, bestela, gordetzeko botoia sakatu emandako nota berresteko, "{0}".
 
 grad3=Kalifikatua
-
+grade.title=Balorazioak
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_eu.properties:269]: Good morning @pablosanroman , could you take a look at these basque strings?

Thanks

File: assignment/api/src/resources/assignment.properties [hunk 17]
```diff
@@ -479,6 +479,7 @@ theisnostudent = There is no student.
 thiasshas = This assignment has no instructions. Please make a correction or click the original button to proceed.
 
 ungra = Ungraded
+unchecked=Unchecked
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment.properties:482]: Are you using this string? I don't see it!

**Review by mpellicer — APPROVED** [review]

**Comment by PabloSanRoman** [comment]
@mpellicer , better like this: 

grade.title=Kalifikazioak

**Comment by mpellicer** [comment]
> @mpellicer , better like this:
> 
> grade.title=Kalifikazioak

Thanks Pablo, we'll take this into consideration for next release which will happen shortly. @JuanDavid102
